src/cloudwatch/modules/configuration/metadatareader.py

```diff
@@ -59,13 +62,31 @@ def _get_metadata(self, request):
                    'http://169.254.169.254/latest/meta-data/placement/availability-zone/' 
                    then the request part is 'latest/meta-data/placement/availability-zone/'.
         """
-        result = self.session.get(self.metadata_server + request, timeout=self._REQUEST_TIMEOUT)
+        headers = {'X-aws-ec2-metadata-token':self.token}
+        result = self.session.get(self.metadata_server + request, timeout=self._REQUEST_TIMEOUT, headers=headers)
+        if result.status_code == codes.unauthorized:
+            self.token = self._get_metadata_token()
+            self._LOGGER.info("Token length: %s " % (str(len(self.token))) )
```

Comment:
why do we log the length here?

Reply:
It is just a record that the data comes from IMDSV2.